src/runtime/internal/Component.ts

```diff
@@ -177,7 +177,7 @@ export function init(component, options, instance, create_fragment, not_equal, p
 }
 
 export let SvelteElement;
-if (typeof HTMLElement === 'function') {
+if (is_function(HTMLElement)) {
```

Comment:
There is a difference between these two if-statements. Doing a `typeof` on something that is not defined is completely legal, e.g doing `typeof window === 'undefined'` if you are in a server context. However, using an undefined variable in a function call is not legal. This is causing the "HTMLElement is not defined" issue 🤔 

Here's a simple REPL to illustrate https://repljs.com/new?local=ee5b7e4f-f398-4ade-8855-3ac0b324812a

Reply:
I know. That's why the tests are failing and I'm wondering why the if statement is needed at all. Is it just here to make the tests pass and makes no sense in the production bundle?

Reply:
Good question! My gut feeling is that this has something to do with SSR. It seems like SvelteElement is a web-component, so maybe it's not created on the server-side. I'm not sure how JSDOM works here (if used in SSR at all), but i guess that the if-statement is there to make sure that we are on the client.

Note that I'm just guessing here

Reply:
Svelte native uses Native Script, and since Native Script does not have HTMLElement, so I believe an error would occur without this if statement.

Reply:
ok, thanks for clearification. I'll revert this change